ketherhomepage/static.go

```diff
@@ -192,34 +192,72 @@ func (w *KetherWatcher) Watch(duration time.Duration) {
 		data := KetherData{BlockNumber: int(blockNumber.Int64()), Ads: ads}
 		json, err := json.Marshal(data)
 		if err != nil {
-			log.Printf("%s: Couldn't marshal ads to json: %v", w.name, err)
+			log.Errorf("%s: Couldn't marshal ads to json: %v", w.name, err)
 			continue
 		}
 
+		log.Printf("%s: Writing JSON", w.name)
 		jsonW := w.jsonObject.NewWriter(w.ctx)
-		jsonW.Write(json)
-		jsonW.Close()
-		log.Printf("%s: Wrote JSON", w.name)
+		_, err = jsonW.Write(json)
+		if err != nil {
+			log.Errorf("%s: Couldn't write json: %v", w.name, err)
+		}
+		err = jsonW.Close()
+		if err != nil {
+			log.Errorf("%s: Couldn't close JSON writer: %v", w.name, err)
+		}
 
+		log.Printf("%s: Writing PNG", w.name)
 		pngW := w.pngObject.NewWriter(w.ctx)
-		png.Encode(pngW, adsImage)
-		pngW.Close()
-		log.Printf("%s: Wrote PNG", w.name)
+		err = png.Encode(pngW, adsImage)
+		if err != nil {
+			log.Errorf("%s: Couldn't write PNG: %v", w.name, err)
+		}
+		err = pngW.Close()
+		if err != nil {
+			log.Errorf("%s: Couldn't close PNG writer: %v", w.name, err)
+		}
 
+		log.Printf("%s: Writing 2x PNG", w.name)
 		png2XW := w.png2XObject.NewWriter(w.ctx)
-		png.Encode(png2XW, adsImage2X)
-		png2XW.Close()
-		log.Printf("%s: Wrote PNG @ 2x", w.name)
+		err = png.Encode(png2XW, adsImage2X)
+		if err != nil {
+			log.Errorf("%s: Couldn't write 2x PNG: %v", w.name, err)
+		}
+		err = png2XW.Close()
+		if err != nil {
+			log.Errorf("%s: Couldn't close 2x PNG: %v", w.name, err)
+		}
 
 		// Set ACLs to public
-		w.jsonObject.ACL().Set(w.ctx, storage.AllUsers, storage.RoleReader)
-		w.pngObject.ACL().Set(w.ctx, storage.AllUsers, storage.RoleReader)
-		w.png2XObject.ACL().Set(w.ctx, storage.AllUsers, storage.RoleReader)
+		log.Printf("%s: Setting ACLs to public", w.name)
```

Comment:
Optional nit: I'd probably write a helper and/or a loop at this point, just because I'd be worried of having typo/outofdate-related bugs from so much replicated code.